### PR TITLE
Bump to v0.9.0

### PR DESCRIPTION
**Pull request by qdequele**

### CODE REVIEW

**Comment by alexisvisco** [comment]
Hi! There is some problems with tests, maybe you should look at it if you have time!

**Comment by qdequele** [comment]
Hi @alexisvisco, thanks for the comment, I've finished the new version. We changed the API a lot in the last version of MeiliSearch. That's why I decided to update this SDK. I would be happy if you take a look if you have time or if you give me a review.

**Comment by qdequele** [comment]
- The SDK doesn't use the same convention as the others. 
- It is not well tested than the other SDKs.

@curquiza what do you think about it?

**Comment by curquiza** [comment]
Hi! I'm not an expert in go, so I might have misunderstood the code but:
- I don't see the optional parameter `primaryKey` for the documents in `AddOrReplace` and `AddOrUpdate`
- Same for the index creation: I don't see the optional `primaryKey` parameter
- Is there a way to set a parameter as optional in go? Because the `name` parameter should be optional during index creation
- @qdequele, we decided together to name the routes to update settings as `updateXXX` and not `setXXX`. Hope it's still correct in go 🤞 
- does this package has a version?

Thanks @alexisvisco for your huge work on this, and thanks @qdequele for the PR :)

**Comment by curquiza** [comment]
@qdequele what do you mean by "same convention as the others", so that I can tell you if it's a big deal or not :)

**Comment by qdequele** [comment]
@curquiza I mean the organization of Struct/Classes and the function names.

**Comment by curquiza** [comment]
For the struct/classes I don't know the best way to do this because the go is not a pure object oriented programming language :)
For the name, I just noticed the settings part. Maybe there are other issues with naming but not a big deal so far.

**Comment by curquiza** [comment]
And @qdequele: you haven't updated the readme ;)

**Comment by qdequele** [comment]
@curquiza 
> I don't see the optional parameter primaryKey for the documents in AddOrReplace and AddOrUpdate
Same for the index creation: I don't see the optional primaryKey parameter
> Is there a way to set a parameter as optional in go? Because the name parameter should be optional during index creation

I think that all fields in a Go structure are optional, there is a way to know if the field exists or not with `val, ok := structField`. I think... And during the serialization/Marshal Go use `json:"primaryKey,omitempty"` to specify that the field can be omitted.

> @qdequele, we decided together to name the routes to update settings as updateXXX and not setXXX. Hope it's still correct in go 🤞

👍 Updated

> does this package has a version?

I think Go use git versioning. I think...

File: readme.md [hunk 1]
```diff
@@ -1,65 +1,286 @@
...
+    "offset": 0,
+    "limit": 1,
+    "processingTimeMs": 0,
+    "query": "prince"
+}
+```
+
+## ⚙️ Development Workflow
+
+If you want to contribute, this sections describes the steps to follow.
```
Comment (curquiza) [inline readme.md]:
```suggestion
If you want to contribute, this section describes the steps to follow.
```

Comment (curquiza) [reply readme.md]: (my bad, I did this mistake everywhere)

File: readme.md [hunk 2]
```diff
@@ -1,65 +1,286 @@
...
+
+#### Add documents <!-- omit in toc -->
+
+```go
+var client = NewClient(Config{
+    Host: "http://127.0.0.1:7700",
+    APIKey: "masterKey"
+})
+
+resp, err := client.Indexes().Create(CreateIndexRequest{
```
Comment (curquiza) [inline readme.md]: How do you get an index if you have already created one?

File: readme.md [hunk 3]
```diff
@@ -1,65 +1,286 @@
...
+## 🔧 Installation
+
+With `go get` in command line:
+```bash
+$ go get github.com/meilisearch/meilisearch-go
+```
+
+### Run MeiliSearch <!-- omit in toc -->
+
+There are many easy ways to [download and run a MeiliSearch instance](https://docs.meilisearch.com/guides/advanced_guides/binary.html#download-and-launch).
```
Comment (curquiza) [inline readme.md]:
```suggestion
There are many easy ways to [download and run a MeiliSearch instance](https://docs.meilisearch.com/guides/advanced_guides/installation.html#download-and-launch).
```

File: readme.md [hunk 4]
```diff
@@ -1,65 +1,286 @@
...
+Thank you for your interest in a MeiliSearch tool! ♥️
+
+### Install Go
+
+Follow the official [tutorial](https://golang.org/doc/install)
+
+### Install dependencies
+
+```bash
+$ go get -v -t -d ./...
```
[the remark this answers is not in the document]
Comment (qdequele) [reply readme.md]: the `./...` is definitely something from Go...

**Pull request merged**